pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -33,16 +34,27 @@ class TrainingTypePlugin(ABC):
     """Base class for all training type plugins that change the behaviour of the training, validation and test-
     loop."""
 
-    def __init__(self, checkpoint_io: Optional[CheckpointIO] = None) -> None:
+    def __init__(
+        self, checkpoint_io: Optional[CheckpointIO] = None, precision_plugin: Optional[PrecisionPlugin] = None
+    ) -> None:
         self._model: Optional[Module] = None
         self._results: Optional[Union[_EVALUATE_OUTPUT, _PREDICT_OUTPUT]] = None
         checkpoint_io = checkpoint_io if checkpoint_io is not None else TorchCheckpointIO()
         self._checkpoint_io = checkpoint_io
+        self._precision_plugin = precision_plugin if precision_plugin is not None else PrecisionPlugin()
 
     @property
     def checkpoint_io(self) -> CheckpointIO:
         return self._checkpoint_io
 
+    @property
+    def precision(self) -> Union[str, int]:
+        return self._precision_plugin.precision
+
+    @property
+    def precision_plugin(self) -> Optional[PrecisionPlugin]:
```

Comment:
```suggestion
    def precision_plugin(self) -> PrecisionPlugin:
```
I believe this should not be optional given the instantiation above.

Reply:
fixes mypy